index.js

```diff
@@ -29,6 +58,15 @@ module.exports.captureExit = function() {
   exit = process.exit;
 
   process.exit = function(code) {
+    if (handlers.length === 0 && lastTime === undefined) {
+      // synchronously exit when `process.exit` is called during a
```

Comment:
I believe this branch is also taken if there where just no handlers attached. The documentation implies this will only happen within an `on('exit'` hook